src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -409,6 +409,27 @@ def reverse_resolve(self, use_case_id: UseCaseID, org_id: int, id: int) -> Optio
         string: str = obj.string
         return string
 
+    def bulk_reverse_resolve(
+        self, use_case_id: UseCaseID, org_id: int, ids: Collection[int]
+    ) -> List[Optional[str]]:
+        """Looks up the stored strings for the given ids.
+
+        Returns a list with the same length as ids parameter. The at position n the result contains
+        either the resolved string for ids[n] or None if ids[n] cannot be resolved.
+        """
+        metric_path_key = METRIC_PATH_MAPPING[use_case_id]
+        table = self._get_table_from_metric_path_key(metric_path_key)
+        try:
+            strings = table.objects.get_many_from_cache(id=id, use_replica=True)
+        except table.DoesNotExist:
+            return [None] * len(ids)
+
+        for obj in strings:
+            if obj:
+                assert obj.organization_id == org_id
+
+        return [obj.string if obj else None for obj in strings]
```

Comment:
How do you know here that the ordering of strings from the query will match the `ids`?